doc/decisions/DR025-UI-delegation-plugin-interface.md

```diff
@@ -0,0 +1,194 @@
+that appropriate UI functionality can be _delegated_ to a plugin, then
+we can have the best of both worlds — a native level integration of
+bespoke interfaces that are optimized for the user's context.
+
+### Aims
+
+The aim of this decision is not to fully design the architecture for UI
+delegation in OpenAssetIO. The aim is to decide on broad architecture,
+with a focus on the fundamentals that must be fixed in place to enable
+future evolution without requiring significant changes by host and
+manager maintainers. That is, the primary focus is on the plugin
+architecture and the (breaking) changes required to support it.
+
+### Prior art
+
+The precursors to OpenAssetIO provide some prior art in the area of UI
+delegation.
+
+#### Katana
+
+[Katana](https://www.foundry.com/products/katana) is a look development
+and lighting tool by Foundry, and its AssetAPI was the original
+inspiration for OpenAssetIO. The AssetAPI and
+[AssetWidgetDelegate](https://learn.foundry.com/katana/Content/tg/asset_management_system_plugin_api/extending_user_interface.html)
+plugin architecture is battle tested and in use in production today.
+
+Katana AssetAPI plugins must be written in C/C++, and the UI delegation
+AssetWidgetDelegate plugin must be written in Python. Therefore, there
+are two independent plugin systems, and the AssetWidgetDelegate plugin
+is linked to the AssetAPI plugin only by a shared unique string
+identifier.
+
+The AssetWidgetDelegate plugin typically accesses the AssetAPI via a
+global singleton and defines functions that provide or modify PyQt
+widgets (Katana is Qt based). The plugin is expected to inherit from a
+`BaseAssetWidgetDelegate` class and override methods that create or
+modify specific types of widgets. For example, creating an asset
+identifier inline control, or customizing an asset browser panel.
+
+#### FnAssetAPI
+
+FnAssetAPI was the direct ancestor of modern OpenAssetIO and was born
+out of experience with Katana's AssetAPI by Foundry and its customers,
+originally back in 2013. Development was in part a collaboration with
+[Ftrack](https://www.ftrack.com/en/) - an off-the-shelf asset management
+system for media creation. This culminated in a public demo showcasing a
+native-level integration of Ftrack into Foundry's Hiero and Nuke at [NAB
+2013](https://youtu.be/E1kpWC_pGbw?t=1212). Development of FnAssetAPI
+continued for a while after, but was ultimately placed on hold, until
+2021 when it formed the basis of OpenAssetIO.
+
+FnAssetAPI was entirely Python-based for most of its life. C++
+functionality was begun but not proven out, and the approach was not
+adopted for OpenAssetIO's C++ architecture.
+
+The UI delegation layer of FnAssetAPI took a slightly different approach
+to Katana. In FnAssetAPI, a single plugin system was used to load both
+the `ManagerInterfaceBase`-derived core interface and a
+`ManagerUIDelegate`-derived UI delegate. That is, a single plugin
```

Comment:
Fwiw, iirc, this was purely done for convenience - as it was possible, in Python, not to have a UI framework as a dependency unless it was needed. The plan for C++ was that they'd be separate due to the noted con that its difficult to manage c++ dependencies at runtime.